### PR TITLE
implement subscriber mutator and membership mutator for etcd version …

### PR DESCRIPTION
**Pull request by jshencode**
1. implement subscriber mutator and membership mutator for etcd version
2. remove watcher for subscriber heartbeat and use only pull mode and hash for assignment calculation to avoid eager recalculation of assignment during subscriber restart

### CODE REVIEW

File: controller/mutators/etcd/membership_mutator.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		etcdClient: etcdClient,
+	}
+}
+
+type membershipMutatorImpl struct {
+	etcdClient *kvstore.EtcdClient
+}
+
+func (mm membershipMutatorImpl) Join(namespace string, instance models.Instance) error {
+	return errors.New("join not supported in etcd version of membership mutator")
```
Comment (shz117) [inline controller/mutators/etcd/membership_mutator.go]: we depend on datanode heart beat instead?

Comment (jshencode) [reply controller/mutators/etcd/membership_mutator.go]: yeah. easiest route i think

Comment (jshencode) [reply controller/mutators/etcd/membership_mutator.go]: it is not through controller, each datanode heartbeat using m3db's heartbeat service

Comment (voyager-dw) [reply controller/mutators/etcd/membership_mutator.go]: let's have some talk to make it clear

**Pull request merged**